Source/WebCore/page/PointerCaptureController.cpp

```diff
@@ -195,7 +201,21 @@ bool PointerCaptureController::preventsCompatibilityMouseEventsForIdentifier(Poi
     return capturingData && capturingData->preventsCompatibilityMouseEvents;
 }
 
-#if ENABLE(TOUCH_EVENTS) && PLATFORM(IOS_FAMILY)
+#if ENABLE(TOUCH_EVENTS) && (PLATFORM(IOS_FAMILY) || PLATFORM(WPE))
+static Ref<PointerEvent> createPointerEvent(const AtomString& type, const PlatformTouchEvent& platformTouchEvent, unsigned index, bool isPrimary, WindowProxy& view, const Touch* touch)
+{
+#if PLATFORM(IOS_FAMILY)
+    if (type.isEmpty())
+        return PointerEvent::create(platformTouchEvent, index, isPrimary, view);
+    return PointerEvent::create(type, platformTouchEvent, index, isPrimary, view);
+#else
+    UNUSED_PARAM(platformTouchEvent);
+    UNUSED_PARAM(index);
+    UNUSED_PARAM(isPrimary);
+
```

Comment:
The empty type happens only for IOS because `PointerCaptureController::dispatchEventForTouchAtIndex()` is called from there omitting the type argument. All other calls pass a type explicitly.

I've added a `ASSERT(!type.isEmpty())` here.